### PR TITLE
implement example lang formatter in java instead of xtend

### PR DESCRIPTION
**Pull request by cdietrich**
implement example lang formatter in java instead of xtend
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.web.example.statemachine/src-gen/org/eclipse/xtext/web/example/statemachine/services/StatemachineGrammarAccess.java [hunk 1]
```diff
@@ -43,7 +43,8 @@
 		//	states+=State*;
 		@Override public ParserRule getRule() { return rule; }
 		
-		//{Statemachine} signals+=Signal* states+=State*
+		//{Statemachine} signals+=Signal*
```
Comment (szarnekow) [inline org.eclipse.xtext.web.example.statemachine/src-gen/org/eclipse/xtext/web/example/statemachine/services/StatemachineGrammarAccess.java:46]: Unrelated: I wonder if we should use the verbatim rule body instead.
Why: The statemachine example grammar looks like being carefully formatted by hand. I suppose this is true for most languages - the grammar is usually written such that the developer perceives it as well formatted. Why do we bother using the formatter here?

File: org.eclipse.xtext.web.example.statemachine/src/org/eclipse/xtext/web/example/statemachine/formatting2/StatemachineFormatter.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+public class StatemachineFormatter extends AbstractJavaFormatter {
+
+	@Inject
+	private StatemachineGrammarAccess ga;
+
+	protected void format(Statemachine statemachine, IFormattableDocument doc) {
+		for (Signal signal : statemachine.getSignals()) {
+			doc.format(signal);
+			doc.append(signal, it -> it.setNewLines(1, 1, 2));
```
Comment (szarnekow) [inline org.eclipse.xtext.web.example.statemachine/src/org/eclipse/xtext/web/example/statemachine/formatting2/StatemachineFormatter.java:33]: @ArneDeutsch @cdietrich Could it become an idiom to extract methods for these things, e.g. 
```
private ... defaultNewLines() {
 ..
}
```
such that we can write
```
doc.append(signal, this::defaultNewLines);
```

File: org.eclipse.xtext.web.example.statemachine/src/org/eclipse/xtext/web/example/statemachine/formatting2/StatemachineFormatter.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		}
+		for (State state : statemachine.getStates()) {
+			doc.format(state);
+			doc.append(state, it -> it.setNewLines(1, 1, 2));
+		}
+	}
+
+	protected void format(Signal signal, IFormattableDocument doc) {
+		ISemanticRegion signalKeyword = regionFor(signal).keyword(ga.getSignalAccess().getSignalKeyword_1());
+		doc.prepend(signalKeyword, it -> it.oneSpace());
```
Comment (szarnekow) [inline org.eclipse.xtext.web.example.statemachine/src/org/eclipse/xtext/web/example/statemachine/formatting2/StatemachineFormatter.java:43]: Do we want to have delegate methods on the super class for the hidden region formatters?
```doc.prepend(signalKeyword, this::oneSpace);``` ?

**Review by szarnekow — APPROVED** [review]

File: org.eclipse.xtext.web.example.statemachine/src/org/eclipse/xtext/web/example/statemachine/formatting2/StatemachineFormatter.java (hunk 3 above)
Comment (cdietrich) [reply org.eclipse.xtext.web.example.statemachine/src/org/eclipse/xtext/web/example/statemachine/formatting2/StatemachineFormatter.java:43]: cc @ArneDeutsch

**Pull request merged**